roles/ansible/tower/config-ansible-tower-ocp-ssh/tasks/ocp-process-ssh-key.yml

```diff
@@ -0,0 +1,38 @@
+---
+
+- name: Check for existing secret
+  command: |
+    oc get secret {{ ssh_key.secret_name | default('ansible-tower-ssh-key') }} \
+      -o=jsonpath='{.metadata.name}' \
+      -n {{ ssh_key.secret_project }}
+  register: secret_check
+  failed_when: secret_check.rc > 1
+
+- name: Check for existing volume mount
+  command: |
+    oc set volume {{ ssh_key.deployment_type | default('deployment')}}/{{ ssh_key.deployment_name | default('ansible-tower') }} \
+    -n {{ ssh_key.secret_project }}
+  register: volume_check
+
+- name: Create a generic ssh key secret from file
+  command: |
+    oc create secret generic {{ ssh_key.secret_name | default('ansible-tower-ssh-key') }} \
+      --from-file={{ ssh_key.src }} \
+      -n {{ ssh_key.secret_project }}
+  when:
+    - secret_check.rc != 0
+
+- name: Mount generic ssh key secret
+  command: |
+    oc set volume {{ ssh_key.deployment_type | default('deployment')}}/{{ ssh_key.deployment_name | default('ansible-tower') }} \
+      --add \
+      --default-mode 0600 \
+      --read-only \
+      --secret-name {{ ssh_key.secret_name | default(ssh_key.dest | basename) }} \
+      --type {{ ssh_key.volume_type | default('secret') }} \
+      --mount-path {{ ssh_key.dest }} \
```

Comment:
From the README, the `dest` variable is supposed to have a default value of `/var/lib/awx/.ssh + src|basename`. I think that's a good solution, so if that could be added here? (or as a set_fact above so other uses of `dest` above works)

Reply:
Yeah, set_fact definitely on the filename definitely makes this cleaner and more DRY.